Add Fastly support (#597)
FriendsOfSymfony/FOSHttpCacheBundle: Resources/doc/reference/configuration/proxy-client.rst

Service identifier of a `AsyncAws\CloudFront\CloudFrontClient` client. More information is available on the
`AWS Async documentation_`. It can not be used with the ``configuration`` option.

.. _configuration_fastly_proxy_client:

Fastly
----------
.. code-block:: yaml

    # config/packages/fos_http_cache.yaml
    fos_http_cache:
        proxy_client:
            fastly:
                service_identifier: '<my-service-identifier>'
                authentication_token: '<my-authentication-token>'
                soft_purge: true

``service_identifier``

> """""""""""""""""""

Reviewer: ```suggestion
"""""""""""""""""""""
```

**type**: ``string``

> Identifier for the Fastly Service you want to purge the cache for.

Reviewer: ```suggestion
Identifier for the Fastly account you want to purge the cache for.
```

Author: Service is the correct term.

You can have a fastly account with different services (VCL configuration), if you want to purge the cdn(cache) you have to specifiy the service you want to do that in.

Reviewer: okay, if that is fastly terminology, lets keep it.

should "service" be lowercase though?

``authentication_token``

> """""""""""""""""

Reviewer: we try to align the number of `"` to the length of the title

```suggestion
"""""""""""""""""""""""
```

**type**: ``string``

Authentication token (API Token) which can be created in the profile section of your account

``soft_purge``

> """""""""""""""""

Reviewer: ```suggestion
""""""""""""""
```

**type**: ``boolean`` **default**: ``true``

> Boolean for doing soft purges or not on tag & URL purging. Soft purges expires the cache unlike hard purge (removal), and allow grace/stale handling within Fastly VCL.

Reviewer: how about:

```suggestion
Whether to expire the cache (true) or whether to directly remove cache entries. Soft purge allows grace/stale handling.
```

(the grace/stale can also be controlled with the cache-control header by the application.)

Author: That is copied from the (FOSHttpCache documentation)[https://foshttpcache.readthedocs.io/en/latest/proxy-clients.html#fastly-client]. I think it would make sense to just copy the description?

Reviewer: fair enough.
